### PR TITLE
wip: feat(vdp): Implement Pipeline Run Logging

### PR DESCRIPTION
**Pull request by tillknuesting**
Because:

Inability to track records during debugging or improvements. Lack of detailed statistics or insights on pipeline runs.

This commit:

Introduces comprehensive logging features for enhanced debugging. Adds tracking of error messages, timestamps, input/output details and execution status.

### CODE REVIEW

File: pkg/db/migration/000022_init.down.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
+BEGIN;
+
+-- Drop indexes
+DROP INDEX IF EXISTS idx_component_runs_completed_time;
```
Comment (joremysh) [inline pkg/db/migration/000022_init.down.sql:4]: these are not needed

File: pkg/handler/pipeline.go [hunk 2]
```diff
@@ -2112,3 +2112,55 @@ func mergeInputsIntoData(inputs []*structpb.Struct, data []*pb.TriggerData) []*p
...
+
+	resp, err := h.service.ListComponentRuns(ctx, "", req, int(req.Page), int(req.PageSize))
+	if err != nil {
+		return nil, status.Error(codes.Internal, "Failed to list component runs")
+	}
+
+	return resp, nil
+}
+
+func validateListPipelineRunsRequest(req *pb.ListPipelineRunsRequest) error {
```
Comment (joremysh) [inline pkg/handler/pipeline.go:2142]: note: I guess they are not needed

File: pkg/worker/workflow.go [hunk 3]
```diff
@@ -358,6 +432,36 @@ func (w *worker) ComponentActivity(ctx context.Context, param *ComponentActivity
 	logger, _ := logger.GetZapLogger(ctx)
 	logger.Info("ComponentActivity started")
 
+	// Define activity options
+	ao := workflow.ActivityOptions{
```
Comment (joremysh) [inline pkg/worker/workflow.go:436]: @donch1989 do you know what this is for?

Comment (joremysh) [reply pkg/worker/workflow.go:436]: oh I guess it's the configuration for a temporal task.
the ctx below does not match the type needed. do u have any idea how to fix that?

Comment (jvallesm) [reply pkg/worker/workflow.go:436]: This is trying to call a Temporal Activity from another Activity, which goes against how Temporal Workflows are structured. It should be 2 separate activities [called from a Workflow](https://community.temporal.io/t/how-to-call-an-activity-from-other-activity/1294). If one depends on the other's data, the Workflow should orchestrate this.

**Comment by tillknuesting** [comment]
Please rebase the branch on  your own work branch. Will delete the fork after 1. September 2024.

**Pull request closed without merging**